test/e2e/azure_machinepool_drain.go

```diff
@@ -145,10 +147,16 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 	Expect(err).NotTo(HaveOccurred())
 	labelNodesWithMachinePoolName(ctx, workloadClusterProxy.GetClient(), amp.Name, ampmls)
 
+	By("adding the AzureMachinePoolMachine finalizer to the machine pool machines")
```

Comment:
Why is the test adding and removing the finalizer? That's something I'd normally expect the controller to do already.

Reply:
I think the test was flaky before because the machine will start to delete before the test has time to check for a successful drain. This was under Cecile's suggestion here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159#issuecomment-1075383407

Reply:
Ah, that makes sense.

Reply:
GitHub wasn't letting me post a comment earlier... I think we should use a different "test machine pool drain" finalizer, because otherwise the test and the controller are racing to remove the same finalizer no?

Reply:
@CecileRobertMichon Good catch! Just added a different finalizer specific for the test: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3503/files#diff-f20fb04862d5b7d8e948f7d513ecb40ec36bf15a4483835d6d7af5ee338c1a69R32